### DISCUSSION TITLE
I want to add my own Keyboard shortcuts (Windows)

### DISCUSSION THREAD

**Discussion by fadikdawoud** [OpenBoard-org/OpenBoard#1129] [Q&A]
I don't know if this is possible without building and compiling Openboard, but I was wondering if it would be possible to edit or add keyboard shortcuts for certain actions in Openboard. 

I know some buttons have their own shortcuts but they are not standard and do not fit my needs. I tried following the Windows build from source instructions on github but sadly no luck with it. 

So maybe there would be away to create a Plugin or something to have the ability to edit and manage your own Keyboard shortcuts.

_PS: if someone was successful with building Openboard from the source files on windows, can you kindly help me out_

**Comment by kaamui** [comment]
Hi ! After 1.7.2 I'll find time to help you in building it on Windows. Sorry for the delay !

**Comment by letsfindaway** [comment]
Starting with version 1.7.2 there will be an unofficial way to configure keyboard shortcuts for actions without recompiling OpenBoard.

**Warning** This is an unofficial feature. Actions may be added or removed in the future. Also the syntax of the configuration file may change. So use it without any guarantee. This feature is only available starting with version 1.7.2.

Locate your `OpenBoardUser.config` file (see [Configuration](https://github.com/OpenBoard-org/OpenBoard/wiki/Configuration)) and open it in a text editor. If not present, add a new section as follows, here to add `Ctrl+#` as shortcut for the hand tool:

```ini
[Shortcut]
actionHand=Ctrl+#, 0, 0
```

You can view the list of actions here in the source code:
https://github.com/OpenBoard-org/OpenBoard/blob/a7685ae0c93bd370994f421be8118b3c1a2a03a7/src/core/UBShortcutManager.cpp#L120-L204

The action names always start with `action`.

On the right hand of the `=` sign there must be a list of three items, separated by comma:
`keyboard shortcut, 0, 0`

`keyboard shortcut` is the textual representation of a key or key combination. Examples are
- `H` the plain letter h
- `Shift+H` upper case H
- `Alt+H`
- `Ctrl+H`

The two `0` after that are foreseen to assign also a mouse or pen button to the shortcut. However the values to choose are not so easy to explain, so please just add the two zeroes after the keyboard shortcut as in the example above.

Save the settings file and restart OpenBoard. If you did everything right, then your shortcuts should work.

Note that if you want to use a key combination which is already in use for another action, then you also have to change that action, too. It is not allowed to assign the same key combination to more than one action.

**Reply by MartinHinteregger** [reply]
Hi 👋

Please forgive my lack of knowledge, but if I would like to try out shortcuts right now, all I need to do is build the dev branch from source, correct?

Thanks!

PS: I really love openboard, thanks everyone for the great work!

**Reply by letsfindaway** [reply]
Principally, yes.But be aware that the next release for Windows will be based on Qt 6 and on a new version of the MSVC redistributables. The build scrips are already updated, but the build instructions are not. So it might be better if you wait a little bit. OpenBoard 1.7.2 is planned to be released this month.

**Reply by MartinHinteregger** [reply]
Thank you for the information and the quick reply!
